### PR TITLE
Add GraphicsMagick

### PR DESCRIPTION
**Pull request by dsendkowski**

### CODE REVIEW

File: components/image/graphicsmagick/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+COMPONENT_NAME=		GraphicsMagick
+COMPONENT_VERSION=	1.3.25
+COMPONENT_PROJECT_URL=	http://www.graphicsmagick.org/	
+COMPONENT_SUMMARY=	A collection of utilities to support file format conversion,image processing, and 2D vector rendering.
+COMPONENT_FMRI=         image/graphicsmagick
+COMPONENT_CLASSIFICATION=Applications/Graphics and Imaging
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:1fae84925a50c1d0d6f64636ffc57b6458dc892e1f181ea5d6bf731936245005
+COMPONENT_ARCHIVE_URL=ftp://ftp.graphicsmagick.org/pub/$(COMPONENT_NAME)/1.3/$(COMPONENT_ARCHIVE)
```
Comment (alarcher) [inline components/image/graphicsmagick/Makefile]: You could add a component major version variable like https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/developer/gcc-7/Makefile#L21 to handle the link.

File: components/image/graphicsmagick/Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+build:		$(BUILD_32_and_64)
+
+install:	$(INSTALL_32_and_64)
+
+test:		$(NO_TEST)
+
+REQUIRED_PACKAGES += codec/jasper
+REQUIRED_PACKAGES += compress/bzip2
+REQUIRED_PACKAGES += image/library/libjpeg6
```
Comment (alarcher) [inline components/image/graphicsmagick/Makefile]: You should compile with libjpeg8-turbo which will become the new default JPEG implementation, see the transition page https://wiki.openindiana.org/oi/JPEG+8

**Review by alarcher — CHANGES_REQUESTED** [review]
Thank you for working on this.
My main comment is about using the new JPEG library,

**Comment by dsendkowski** [comment]
OK, all the requested changes have been made.

File: components/image/graphicsmagick/Makefile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+#
+# Copyright (c) 2017 Dariusz Sendkowski
+#
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		GraphicsMagick
+COMPONENT_VERSION=	1.3.25
+COMPONENT_MAJOR_VERSION= $(shell echo $(COMPONENT_VERSION) | $(GSED) -e 's/\([0-9]\+\.[0-9]\+\).*/\1/')
+COMPONENT_PROJECT_URL=	http://www.graphicsmagick.org/
```
Comment (pyhalov) [inline components/image/graphicsmagick/Makefile]: Better do something like 
COMPONENT_MJR_VERSION=  1.3
COMPONENT_MNR_VERSION=  25
COMPONENT_VERSION=      $(COMPONENT_MJR_VERSION).$(COMPONENT_MNR_VERSION)

**Comment by dsendkowski** [comment]
Ok, `COMPONENT_MAJOR_VERSION` set manually.

File: components/image/graphicsmagick/graphicsmagick.p5m [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+file path=usr/include/GraphicsMagick/magick/transform.h
+file path=usr/include/GraphicsMagick/magick/type.h
+file path=usr/include/GraphicsMagick/magick/utility.h
+file path=usr/include/GraphicsMagick/magick/version.h
+file path=usr/include/GraphicsMagick/wand/drawing_wand.h
+file path=usr/include/GraphicsMagick/wand/magick_wand.h
+file path=usr/include/GraphicsMagick/wand/pixel_wand.h
+file path=usr/include/GraphicsMagick/wand/wand_api.h
+file path=usr/include/GraphicsMagick/wand/wand_symbols.h
+file path=usr/lib/$(MACH64)/GraphicsMagick-$(COMPONENT_VERSION)/config/delegates.mgk
```
Comment (pyhalov) [inline components/image/graphicsmagick/graphicsmagick.p5m:115]: Do I miss something or you've forgotten to include any libraries ?

Comment (pyhalov) [reply components/image/graphicsmagick/graphicsmagick.p5m:115]: --enable-shared ?

Comment (dsendkowski) [reply components/image/graphicsmagick/graphicsmagick.p5m:115]: Yes, you're right. They were disabled by default. I hope it's fine now.

File: components/image/graphicsmagick/Makefile [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	Copyright.txt
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/configure.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+CFLAGS   += $(JPEG_CPPFLAGS) $(JPEG_CFLAGS)
+LDFLAGS  += $(JPEG_LDFLAGS)
+CONFIGURE_OPTIONS+= --enable-shared
```
Comment (pyhalov) [inline components/image/graphicsmagick/Makefile]: Perhaps, insert empty line?
Otherwise, LGTM

**Pull request merged**